syllabus/curricula/annealing_box.py

```diff
@@ -48,9 +48,13 @@ def sample(self, k: int = 1) -> Union[List, Any]:
         Sample k tasks from the curriculum.
         """
         # Linear annealing from start_values to end_values
-        annealed_values = (
-                self.start_values + (self.end_values - self.start_values) *
-                np.minimum(self.current_step, self.total_steps) / self.total_steps
-        )
-
-        return [annealed_values.copy() for _ in range(k)]
\ No newline at end of file
+        if self._should_use_startup_sampling():
+            return self._startup_sample(k)
+        else:
+            print("AnnealingBoxCurriculum")
+            annealed_values = (
+                    self.start_values + (self.end_values - self.start_values) *
+                    np.minimum(self.current_step, self.total_steps) / self.total_steps
+            )
+            self.sampled_tasks += k
```

Comment:
Should this be here? Doesn't `self._startup_sample` track self.sampled_tasks for you? I don't think we should need to modify it directly in the `sample` function.